specs/order_spec.rb

```diff
@@ -76,34 +76,99 @@
       result.must_equal true
     end
   end
+
+  describe "#remove_product" do
+    it "Decreases the number of products" do
+      products = { "banana" => 1.99, "cracker" => 3.00, "salad" => 4.25 }
+      before_count = products.count
+      order = Grocery::Order.new(1337, products)
+
+      order.remove_product("salad")
+      expected_count = before_count - 1
+      order.products.count.must_equal expected_count
+    end
+
+    it "Is removed from the collection of products" do
+      products = { "banana" => 1.99, "cracker" => 3.00, "salad" => 4.25 }
+      order = Grocery::Order.new(1337, products)
+
+      order.remove_product("salad")
+      order.products.include?("salad").must_equal false
+    end
+
+    it "Returns false if the product is not present" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+
+      order = Grocery::Order.new(1337, products)
+      before_total = order.total
+
+      result = order.remove_product("salad")
+      after_total = order.total
+
+      result.must_equal false
+      before_total.must_equal after_total
+    end
+  end
 end
 
-# TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+describe "Order Wave 2" do
   describe "Order.all" do
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
-      # Useful checks might include:
-      #   - Order.all returns an array
-      #   - Everything in the array is an Order
-      #   - The number of orders is correct
-      #   - The ID and products of the first and last
-      #       orders match what's in the CSV file
-      # Feel free to split this into multiple tests if needed
+      # checks that .all returns array
+      Grocery::Order.all.must_be_kind_of Array
+    end
+
+    it "All items are Orders" do
+      # checks that all items returned by .all are Orders
+      orders = Grocery::Order.all
+      orders.each do |item|
+        item.must_be_instance_of Grocery::Order
+      end
+    end
+
+    it "Order total is correct" do
+      # checks order total
+      total_orders = Grocery::Order.all.length
+      total_orders.must_equal 100
+    end
+
+    it "ID and product of the first order is correct" do
+      # tests ID and products for first order
+      first_order = [ 1, {"Slivered Almonds"=>22.88, "Wholewheat flour"=>1.93, "Grape Seed Oil"=>74.9} ]
```

Comment:
The idea behind this test is solid, but I'm curious why you store the expected values in an array. Why not have 2 local variables, `expected_id` and `expected_products`?